server/src/main/java/io/crate/metadata/settings/session/SessionSettingRegistry.java

```diff
@@ -155,10 +173,29 @@ public Map<String, SessionSetting<?>> settings() {
     }
 
     private static String[] objectsToStringArray(Object[] objects) {
-        String[] strings = new String[objects.length];
+        ArrayList<String> argumentList = new ArrayList<>();
         for (int i = 0; i < objects.length; i++) {
-            strings[i] = DataTypes.STRING.implicitCast(objects[i]);
+            String str = DataTypes.STRING.implicitCast(objects[i]).trim();
+            argumentList.addAll(Arrays.stream(str.split(","))
+                              .map(String::trim)
+                              .collect(Collectors
+                              .toCollection(ArrayList::new)));
+        }
+        return argumentList.toArray(String[]::new);
+    }
+
+    private static void validateDateStyleFrom(String... strings) {
+        String dateStyle;
+        for (String s : strings) {
+            dateStyle = s.toUpperCase(Locale.ENGLISH);
+            switch (dateStyle) {
```

Comment:
Shouldn't we have an inverted check and only allow `ISO` and `MDY` here?

Reply:
The inverted check is much bigger as the date orderings include
- `MDY` with the synonyms `NonEuro`, `NonEuropean` and `US`
- `DMY` with the synonyms `Euro` and `European`
- `YMD` 

i.e. the check is only for date time output formatting, which setting would have effect on CrateDB.

Reply:
The PostgreSQL `DateStyle` has two arguments that are not entirely independent.

The date style argument:
- `ISO`
- `GERMAN`
- `SQL`
- `POSTGRES`

and the order argument:
- `MDY` with the synonyms `NonEuro`, `NonEuropean` and `US`
- `DMY` with the synonyms `Euro` and `European`
- `YMD` 

---

The later (order argument) has no effect in PostgreSQL for the formatting of timestamps:
````
postgres=# SET DateStyle TO ISO, MDY;
SET
postgres=# SELECT now();
              now              
-------------------------------
 2022-10-10 14:15:37.554191+02
(1 row)

postgres=# SET DateStyle TO ISO, DMY;
SET
postgres=# SELECT now();
              now              
-------------------------------
 2022-10-10 14:15:44.064077+02
(1 row)

postgres=# SET DateStyle TO ISO, YMD;
SET
postgres=# SELECT now();
              now              
-------------------------------
 2022-10-10 14:15:51.774385+02
(1 row)
```

---

However PostgreSQL also allows to provide either of those arguments independently in any order

```sql
SET DateStyle TO ISO;
SET DateStyle TO DMY;
SET DateStyle TO ISO, DMY;
SET DateStyle TO DMY, ISO;
```

---

So the idea was to allow any order arguments (as they don't have any effect in CrateDB yet) and raise an exception for trying to set the date format style argument (i.e. `ISO`, `SQL`, ...). With the main reason, that many tools are reseting the datestyle to `ISO, DMY`. E.g. also PostgresqlJDBC, which is specifically only checking for the `ISO` part, but doesn't care for the ordering

Reply:
But what happens if you set to something invalid?
`SET datestyle TO MYISO`
`SET datestyle TO ISO,YDM`
In postgres:
```
matriv=# SET DateStyle TO MYISO;
ERROR:  invalid value for parameter "DateStyle": "myiso"
DETAIL:  Unrecognized key word: "myiso".
matriv=# SET DateStyle TO ISO,YDM;
ERROR:  invalid value for parameter "DateStyle": "iso, ydm"
DETAIL:  Unrecognized key word: "ydm".
```

Reply:
I added validation for the date ordering to catch all invalid parameter values